### PR TITLE
Feature/89-Replace-SampleAgent-by-MovieBot-in-`run_simulation`

### LINKED ISSUE #89: Replace SampleAgent by MovieBot in `run_simulation`
Use MovieBot agent for minimal example. Add section in README with running instructions.

### PR DESCRIPTION
**Pull request by NoB0**
Fixes #89 #73 

Depends on #88 

After merging, do we want to remove the `SampleAgent`?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,10 +21,53 @@ UserSimCRS follows the architecture of a typical task-based dialogue system, whi
...
+```shell
+python -m pip install -r requirements/requirements.txt
+```
+To work on the documentation you also need to install other dependencies:
+```shell
+python -m pip install -r requirements/docs_requirements.txt
+```
+## Usage 
+
+A YAML configuration file is necessary to start the MovieBot, see [default configuration](config/default/config_default.yaml) as example.
```
Comment (kbalog) [inline README.md:37]: Nit: 
- ", see" -> "; see"
- "as example" => "for an example"

File: README.md [hunk 2]
```diff
@@ -19,10 +21,53 @@ UserSimCRS follows the architecture of a typical task-based dialogue system, whi
...
+The recommended version of Python is 3.9.  
+The easiest way to install UserSimCRS and all of its dependencies is by using pip:
+```shell
+python -m pip install -r requirements/requirements.txt
+```
+To work on the documentation you also need to install other dependencies:
+```shell
+python -m pip install -r requirements/docs_requirements.txt
+```
+## Usage
```
Comment (kbalog) [inline README.md:35]: Nit: empty line above

File: README.md [hunk 3]
```diff
@@ -19,10 +21,53 @@ UserSimCRS follows the architecture of a typical task-based dialogue system, whi
...
+
+The recommended version of Python is 3.9.  
+The easiest way to install UserSimCRS and all of its dependencies is by using pip:
+```shell
+python -m pip install -r requirements/requirements.txt
+```
+To work on the documentation you also need to install other dependencies:
+```shell
+python -m pip install -r requirements/docs_requirements.txt
+```
```
Comment (kbalog) [inline README.md:34]: Nit: empty lines above and below ```

File: README.md [hunk 4]
```diff
@@ -19,10 +21,53 @@ UserSimCRS follows the architecture of a typical task-based dialogue system, whi
...
+## Usage 
+
+A YAML configuration file is necessary to start the MovieBot, see [default configuration](config/default/config_default.yaml) as example.  
+Run the following command to start the simulation:
+```shell
+python -m usersimcrs.run_simulation -c <path_to_config.yaml>
+```
```
Comment (kbalog) [inline README.md:41]: Nit: empty lines above and below ```

File: README.md [hunk 5]
```diff
@@ -19,10 +21,53 @@ UserSimCRS follows the architecture of a typical task-based dialogue system, whi
...
+### Example
+
+This example shows how to run simulation using the default configuration and the [IAI MovieBot](https://github.com/iai-group/MovieBot).
```
Comment (kbalog) [inline README.md:44]: "...and the IAI MovieBot" => "... and the IAI MovieBot as the conversational agent"

File: README.md [hunk 6]
```diff
@@ -19,10 +21,53 @@ UserSimCRS follows the architecture of a typical task-based dialogue system, whi
...
+  * Start the IAI MovieBot locally: `python -m run_bot -c config/moviebot_config_no_integration.yaml`
+
+Note: you need to update the parameter `moviebot_uri` in the configuration in case MovieBot does not run on the default URI (i.e., `http://127.0.0.1:5001`).
+
+2. Run simulation
+
+```shell
+python -m usersimcrs.run_simulation -c config/default/config_default.yaml
+```
+
```
Comment (kbalog) [inline README.md:60]: Add a line after this stating what this should produce (e.g., X number of files under data/runs).

File: README.md [hunk 7]
```diff
@@ -19,10 +21,53 @@ UserSimCRS follows the architecture of a typical task-based dialogue system, whi
...
 ## Contributors
 
-(Alphabetically ordered by last name) Jafar Afzali, Krisztian Balog, Aleksander Drzewiecki and Shuo Zhang
+(Alphabetically ordered by last name)
+
+  * Jafar Afzali 
+  * Krisztian Balog 
+  * Nolwenn Bernard 
+  * Aleksander Drzewiecki 
+  * Shuo Zhang
```
Comment (kbalog) [inline README.md:73]: Add years as well, similar to DialogueKit

File: config/default/config_default.yaml [hunk 8]
```diff
@@ -1,10 +1,12 @@
-agent_id: "SampleAgent"
-output_name: "sample_agent"
+agent_id: "MovieBotAgent"
+output_name: "moviebot_agent"
```
Comment (kbalog) [inline config/default/config_default.yaml:2]: How about?
```
agent_id: "IAI MovieBot"
output_name: "moviebot"
```

File: usersimcrs/run_simulation.py [hunk 9]
```diff
@@ -41,17 +43,16 @@
...
+def main(config: confuse.Configuration, agent: Agent) -> None:
     """Executes the specified configuration.
 
     Loads domain and interaction model. Initializes agent and user. Runs the
     simulation.
 
     Args:
         config: Configuration generated from YAML configuration file.
+        agent: Conversational agent.
     """
```
Comment (kbalog) [inline usersimcrs/run_simulation.py:55]: What do you think about adding a type check in the beginning for `agent`?

File: usersimcrs/run_simulation.py [hunk 10]
```diff
@@ -205,8 +206,8 @@ def load_config(args: argparse.Namespace) -> confuse.Configuration:
     return config
 
 
-def get_intent_classifier(config: confuse.Configuration) -> IntentClassifier:
-    """Returns intent classifier model.
+def get_NLU(config: confuse.Configuration) -> IntentClassifier:
+    """Returns a NLU component.
```
Comment (kbalog) [inline usersimcrs/run_simulation.py:210]: Nit: "a NLU" => "an NLU" (also below)

File: config/default/config_default.yaml [hunk 11]
```diff
@@ -1,10 +1,12 @@
...
+# moviebot_uri is only needed to run the usage example.
+moviebot_uri: "http://127.0.0.1:5001"
```
Comment (kbalog) [inline config/default/config_default.yaml:4]: What about calling it `agent_uri` generally? The comment could state that by default it's assumed that the agent has a http API.

File: usersimcrs/run_simulation.py [hunk 12]
```diff
@@ -283,16 +286,30 @@ def load_rasa_diet_classifier(
...
+    return intent_classifier
 
 
 if __name__ == "__main__":
     args = parse_args()
     config = load_config(args)
-    main(config)
+
+    # Defines the agent for the simulation.
+    # By default, a local moviebot is used. See TODO for more details.
```
Comment (kbalog) [inline usersimcrs/run_simulation.py:303]: Which TODO is being referred to here?

File: usersimcrs/run_simulation.py [hunk 13]
```diff
@@ -283,16 +286,30 @@ def load_rasa_diet_classifier(
...
+    moviebot_uri = config["moviebot_uri"].get()
+    try:
+        response = requests.get(moviebot_uri)
+        assert response.status_code == 200
+        agent = MovieBotAgent(agent_id="MovieBotTester", uri=moviebot_uri)
+    except requests.exceptions.RequestException:
+        raise RuntimeError(
+            f"Connection refused to {moviebot_uri}. Please check that "
+            "moviebot is running at this address. See the full traceback above."
```
Comment (kbalog) [inline usersimcrs/run_simulation.py:312]: replace `moviebot_uri` with `agent_uri`. In the exception text, "Please check that the conversational agent is ..."

File: usersimcrs/simulator/agenda_based/agenda_based_simulator.py [hunk 14]
```diff
@@ -54,9 +55,11 @@ def _generate_response(self, agent_utterance: Utterance) -> Utterance:
...
             User utterance.
         """
-        self.generate_response(agent_utterance)
+        return self.generate_response(agent_utterance)
 
-    def generate_response(self, agent_utterance: Utterance) -> Utterance:
+    def generate_response(
+        self, agent_utterance: Utterance
+    ) -> AnnotatedUtterance:
         """Generate response to the agent utterance.
```
Comment (kbalog) [inline usersimcrs/simulator/agenda_based/agenda_based_simulator.py:63]: Nit: "Generates a response ... agent's ..."

**Review by kbalog — APPROVED** [review]
LGTM.

Indeed, the SampleAgent may be removed.

File: README.md (hunk 3 above)
Comment (NoB0) [reply README.md:34]: Done

File: README.md (hunk 1 above)
Comment (NoB0) [reply README.md:37]: Done.

File: README.md (hunk 2 above)
Comment (NoB0) [reply README.md:35]: Done.

File: README.md (hunk 5 above)
Comment (NoB0) [reply README.md:44]: Done.

File: config/default/config_default.yaml (hunk 8 above)
Comment (NoB0) [reply config/default/config_default.yaml:2]: Done.

File: config/default/config_default.yaml (hunk 11 above)
Comment (NoB0) [reply config/default/config_default.yaml:4]: Good point. Done.

File: usersimcrs/run_simulation.py (hunk 10 above)
Comment (NoB0) [reply usersimcrs/run_simulation.py:210]: Done.

File: usersimcrs/run_simulation.py (hunk 12 above)
Comment (NoB0) [reply usersimcrs/run_simulation.py:303]: I forgot to add the pointer to usage example. Done now.

File: usersimcrs/simulator/agenda_based/agenda_based_simulator.py (hunk 14 above)
Comment (NoB0) [reply usersimcrs/simulator/agenda_based/agenda_based_simulator.py:63]: Done.

File: README.md (hunk 6 above)
Comment (NoB0) [reply README.md:60]: Done.

File: usersimcrs/run_simulation.py (hunk 9 above)
Comment (NoB0) [reply usersimcrs/run_simulation.py:55]: Done.

**Pull request merged**